### DISCUSSION TITLE
python@3.10

### DISCUSSION THREAD

**Discussion by luckman212** [Homebrew/discussions#2944] [Everyday usage]
After reading [Python 3.9.x & 3.10.x #91421](https://github.com/Homebrew/homebrew-core/issues/91421) and [python@3.10: migrate aliases - PR #87075](https://github.com/Homebrew/homebrew-core/pull/87075) I find myself still scratching my head on the best way to have my scripts make use of Python 3.10 **today**.

Should I ...
- add `/opt/homebrew/opt/python@3.10/bin` to the front of `PATH` ?
- symlink `/usr/local/bin/python310` to `/opt/homebrew/opt/python@3.10/bin/python3` ?
- use a custom shebang e.g. `#!/usr/local/bin/python310` ?
- use `virtualenv`s for everything?

For now I've settled on the custom shebang + symlink, seems the least prone to causing side-effects but curious what others are doing. I mainly want to use 3.10 to be able to use the new [Structural Pattern Matching](https://docs.python.org/3.10/whatsnew/3.10.html#pep-634-structural-pattern-matching) aka `match...case` statements.

**Comment by SMillerDev** [comment]
Prepend the path or use `#!/usr/bin/env python3.10`
